lambda/dockerimagebuilder/__init__.py

```diff
@@ -53,20 +54,38 @@ def handler(event: Dict[str, Any], context) -> None:  # type: ignore [no-untyped
     rendered_userdata = rendered_userdata.replace("{{BASE_IMAGE}}", shlex.quote(base_image))
     rendered_userdata = rendered_userdata.replace("{{MOUNTS3_DEB_URL}}", shlex.quote(mounts3_deb_url))
     rendered_userdata = rendered_userdata.replace("{{ECR_URI}}", shlex.quote(os.environ["LISA_ECR_URI"]))
-    rendered_userdata = rendered_userdata.replace("{{IMAGE_ID}}", str(uuid.uuid4()))
+    rendered_userdata = rendered_userdata.replace("{{IMAGE_ID}}", image_tag)
 
     print("Creating instance with userdata: ")
     print(rendered_userdata)
     try:
-        ec2_resource.create_instances(
+        instances = ec2_resource.create_instances(
             ImageId=ami_id,
             MinCount=1,
             MaxCount=1,
             InstanceType="m5.large",
             UserData=rendered_userdata,
             IamInstanceProfile={"Arn": os.environ["LISA_INSTANCE_PROFILE"]},
             BlockDeviceMappings=[{"DeviceName": "/dev/xvda", "Ebs": {"VolumeSize": 32}}],
+            TagSpecifications=[
+                {
+                    "ResourceType": "instance",
+                    "Tags": [
+                        {
+                            "Key": "Name",
+                            "Value": f"LisaDockerImageBuilder_{image_tag}"
+                        },
+                        {
+                            "Key": "lisa_temporary_instance",
+                            "Value": "true"
+                        }
+                    ]
+                }
+            ]
         )
+        return {
+            "instance_id": instances[0].instance_id,
+            "image_tag": image_tag
+        }
     except ClientError as e:
```

Comment:
I'm not sure which errors we're expecting, but I think we can just let them boil out to the state machine. From the state machine definition after it's deployed, it automatically handles some form of sdk errors, but I'm not 100% sure if those are strictly lambda or if they're any sdk call:

```json
      "Retry": [
        {
          "ErrorEquals": [
            "Lambda.ClientExecutionTimeoutException",
            "Lambda.ServiceException",
            "Lambda.AWSLambdaException",
            "Lambda.SdkClientException"
          ],
          "IntervalSeconds": 2,
          "MaxAttempts": 6,
          "BackoffRate": 2
        }
      ],
```